lerobot/common/policies/diffusion/modeling_diffusion.py

```diff
@@ -67,15 +67,32 @@ def __init__(
         self.diffusion = DiffusionModel(config)
 
     def reset(self):
-        """
-        Clear observation and action queues. Should be called on `env.reset()`
-        """
+        """Clear observation and action queues. Should be called on `env.reset()`"""
         self._queues = {
             "observation.image": deque(maxlen=self.config.n_obs_steps),
             "observation.state": deque(maxlen=self.config.n_obs_steps),
             "action": deque(maxlen=self.config.n_action_steps),
         }
 
+    def _preprocess_batch_keys(self, batch: dict[str, Tensor], train_mode: bool = False):
+        """Check that the keys can be handled by this policy and standardize the image key.
+
+        This should be run after input normalization.
+        """
+        assert "observation.state" in batch
+        # There should only be one image key.
```

Comment:
FYI: Actually this is too dynamic. I think I should just check `config.input_shapes` and make sure the key matches and there are not more.